geopandas/geodataframe.py

```diff
@@ -214,10 +214,33 @@ def __setattr__(self, attr, val):
 
     def _get_geometry(self):
         if self._geometry_column_name not in self:
-            raise AttributeError(
-                "No geometry data set yet (expected in"
-                " column '%s'.)" % self._geometry_column_name
-            )
+            if self._geometry_column_name is None:
```

Comment:
Updated in line with https://github.com/geopandas/geopandas/issues/2133#issuecomment-1024111760,
but explicitly handling the different possible cases.